### ISSUE TITLE
Sitemap: additionalWhere prevents finding pages in deeper level in pagetree

### ISSUE THREAD

**Issue by mhuijser** [Yoast/Yoast-SEO-for-TYPO3#215]
* [x ] I've read and understood the [contribution guidelines](https://github.com/Yoast/Yoast-SEO-for-TYPO3/blob/master/.github/CONTRIBUTING.md).
* [x ] I've searched for any related issues and avoided creating a duplicate issue.

### Please give us a description of what happened.

I'm trying to  create a news sitemap that only contains my "Articles". Therefore i'm using the typoscript settings to filter pages for another doktype=70  as described in https://docs.typo3.org/typo3cms/extensions/yoast_seo/Configuration/Sitemap/Index.html 

```
plugin.tx_yoastseo.sitemap {
    config {
        pages {
            additionalWhere = AND doktype IN(1,33)
        }
        articles {
            partialName = Pages
            table = pages
            additionalWhere = AND doktype IN(70)
        }
    }
}
```

### Please describe what you expected to happen and why.

I expected to get all pages with doktype=70, but instead i got only 1 entry: the root-page (uid=1) which ironically  doesn't have doktype 70 at all ;-)

### How can we reproduce this behavior?
1. Create a few "pages" with another doktype than "1" in a deeper level in the pagetree
2. Configure a sitemap in typoscript as described in https://docs.typo3.org/typo3cms/extensions/yoast_seo/Configuration/Sitemap/Index.html  to obtain pages with the specified doktype
3. Test the sitemap 

### Technical info
* TYPO3 version: 8.7.19
* Yoast SEO version: 3.0.4

**Comment by ayacoo** [comment]
@mhuijser We need this option, too.

My hints:

- Your current code doens't run under TYPO3 7.6 because of the query builder. Yoast 3.x supports 7.6 (composer.json)
- I would define "recursive" as option not as hardcoded 99

My wishes:
- We have also a rootPid option for "other" plugins.
- We implement a recursive option for pages and "other" plugins

What do you think?

**Comment by haassie** [comment]
Fixed by #236
